src/main/java/org/apache/maven/plugins/pmd/exec/PmdExecutor.java

```diff
@@ -429,7 +430,7 @@ private void writeFormattedReport( Report report )
             throws MavenReportException
     {
         Renderer renderer = createRenderer( request.getFormat(), request.getOutputEncoding() );
-        writeReport( report, renderer, request.getFormat() );
+        writeReport( report, renderer, renderer.defaultFileExtension() );
```

Comment:
Here a NPE might occur (and it does in the integration test multi-module), if the request format "none" is used, which means "no report" - and then there is no renderer.
Since write report is a private method, I'll change that to determine the file extension after the check that we have a renderer.